packages/@vuepress/core/src/page/resolvePageData.ts

```diff
@@ -19,14 +19,17 @@ export const resolvePageData = async ({
     frontmatter,
     excerpt,
     headers,
+    meta: {},
   }
 
   // plugin hook: extendsPageData
   const extendsPageData = await app.pluginApi.hooks.extendsPageData.process(
     page,
     app
   )
-  extendsPageData.forEach((item) => Object.assign(data, item))
+  extendsPageData.forEach((item) =>
+    Object.assign(data, { item, meta: { ...data.meta, ...item.meta } })
+  )
```

Comment:
Here I deep merge `meta` key to avoid the frontter edits being coverd.

But, I think there are issues with this design. I will open a new issue to explain why there are problems.